app/controllers/decidim/analytics/admin/analytics_controller.rb

```diff
@@ -4,11 +4,25 @@ module Decidim
   module Analytics
     module Admin
       class AnalyticsController < Analytics::Admin::ApplicationController
-
         def index
-          @server_address = Rails.application.secrets.dig(:matomo, :server_address)
-          @site_id = Rails.application.secrets.dig(:matomo, :site_id)
-          @token_auth = Rails.application.secrets.dig(:matomo, :token_auth)
+          @server_address = matomo_config[:server_address]
+          @site_id = matomo_config[:site_id]
+          @token_auth = matomo_config[:token_auth]
+
+          if missing_config?
+            flash.now[:alert] = t("decidim.analytics.admin.missing_config")
+          end
+        end
+
+        private
+
+        def matomo_config
+          @matomo_config ||= Rails.application.credentials.dig(:matomo) ||
```

Comment:
In decidim we have never used credentials and we won't use it as it makes difficult to manage the instances. Use the secrets instead. Or, even better, the module configuration with env vars. As the secrets won't be available from decidim 0.31